File: src/commands/whoami.ts
import { Command, flags } from "@oclif/command";
import { readConfigFile } from "../../app/config";
import { PRIMARY_COLOR } from "../../constants";
import { bold, withPrimary } from "../../design/layout";

export default class WhoamI extends Command {
  static description = "Display the current user.";

  static flags = {
    help: flags.help({ char: "h" }),
    username: flags.boolean({
      char: "u",
      description: "just show the username",
    }),
  };

  async run() {
    const { flags } = this.parse(WhoamI);

    const username = readConfigFile("name");
    const email = readConfigFile("user");

    if (username) {
      switch (flags.username) {
        case true:
          this.log(username);

          break;

        default:
          console.log(`👊 Hi ${withPrimary(username)} <${bold(email)}>`);
      }
    }
  }
}

File: app/whoami.go
package app

import (
	"fmt"

	"github.com/spf13/cobra"
	"github.com/charmbracelet/lipgloss"
	"github.com/scmn-dev/secman/constants"
	"github.com/scmn-dev/secman/internal/config"
)

func WhoamiCMD() *cobra.Command {
	cmd := &cobra.Command{
		Use:   "whoami",
		Short: "Display the current user.",
		Long:  "Display the current user.",
		RunE: func(cmd *cobra.Command, args []string) error {
			s := lipgloss.NewStyle().PaddingLeft(2)

			primary := lipgloss.NewStyle().Foreground(lipgloss.Color(constants.PRIMARY_COLOR))
			bold := lipgloss.NewStyle().Bold(true)

			fmt.Println(s.Render("\n👊 Hi ") + primary.Render(config.Config("config.name")) + " <" + bold.Render(config.Config("config.user")) + ">")

			return nil
		},
	}

	return cmd
}
